### DISCUSSION TITLE
Question: How to access wiimote input data [DSU Client] to make another script

### DISCUSSION THREAD

**Discussion by martinprad0** [v1993/linuxmotehook2#14] [Q&A] [answered]
I have built linuxmotehook2 on my laptop and want to access the wiimote controller input from the DSU Client. My goal is to write a bash script that makes the wiimote pointer my desktop's mouse and maps the wiimote buttons to other keys.

![image](https://github.com/v1993/cemuhook-protocol/assets/66460425/be4f9d31-72df-49c2-906b-9b84f3db52f1)

The question is how can I access the data in `localhost:26760` to make this happen.

**Comment by martinprad0** [comment]
I'm not very familiar with the GitHub discussion system. With a second though, I believe that the issues section was not the correct place to start this question. I'm really sorry.

**Comment by v1993** [comment, marked as answer]
I'd recommend looking into using xwiimote library directly instead of relying on this project as middleware:

* linuxmotehook2 does not expose wiimote's IR pointer if that's what you want to use - only motion/buttons.
* xwiimote library is likely quite a bit easier to use than DSU protocol. Neither is (reasonably) usable from bash, though - which is just a poor fit for this task overall.
* To elaborate on the previous point - xwiimote provides an interface that's purposefully easy to use; with DSU protocol you'll have to encode and parse binary packets yourself.

I'd suggest either using C/C++ interface directly, using this project's Vala bindings, or [Lua bindings](https://github.com/v1993/lua-xwiimote) I wrote a while ago (plus probably `lua-posix` for file descriptor polling). There are also ancient official bindings for Python 2, but you shouldn't use them for obvious reasons.

**Reply by martinprad0** [reply]
Thank you! I'll investigate more.

**Reply by v1993** [reply]
I think that for a personal project like this using Lua bindings would be the best choice; while somewhat harder to redistribute it's much quicker to develop for. Otherwise, I'd go with Vala (which linuxmotehook2 is written in).

Feel free to ask further questions!

**Reply by v1993** [reply]
It may be worth noting that there's an old X11 module that allows to use wiimote like you've described. Unfortunately, one has to edit X config to turn it off/on and I've found it to be very crash-prone, so can't recommend using it in practice.

**Reply by martinprad0** [reply]
Hi, I know it's been some time since we had this discussion, but I wanted to thank you for the support. I recently made a (Python) script using xWiimote for using the wiimote as a mouse cursor with uinput, and I wouldn't have been able to do this without your guidance.

**Reply by v1993** [reply]
Glad to hear I was able to help!
